CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SyntaxRewriterFile.swift

```diff
@@ -0,0 +1,303 @@
+      ///         `visit(_:)` methods will not be called for this node and the
+      ///         visited node will be replaced by the returned node in the
+      ///         rewritten tree.
+      open func visitAny(_ node: Syntax) -> Syntax? {
+        return nil
+      }
+      """)
+    
+    FunctionDecl("""
+    /// The function called after visiting the node and its descendents.
+    ///   - node: the node we just finished visiting.
+    open func visitPost(_ node: Syntax) {}
+    """)
+    
+    FunctionDecl("""
+      /// Visit any Syntax node.
+      ///   - Parameter node: the node that is being visited
+      ///   - Returns: the rewritten node
+      public func visit(_ node: Syntax) -> Syntax {
+        return visit(node.data)
+      }
+      """)
+    
+    FunctionDecl("""
+      public func visit<T: SyntaxChildChoices>(_ node: T) -> T {
+        return visit(Syntax(node)).cast(T.self)
+      }
+      """)
+    
+    for baseKind in SYNTAX_BASE_KINDS.filter { !["Syntax", "SyntaxCollection"].contains($0) } {
+      FunctionDecl("""
+        /// Visit any \(raw: baseKind)Syntax node.
+        ///   - Parameter node: the node that is being visited
+        ///   - Returns: the rewritten node
+        public func visit(_ node: \(raw: baseKind)Syntax) -> \(raw: baseKind)Syntax {
+          return visit(node.data).cast(\(raw: baseKind)Syntax.self)
+        }
+        """)
+    }
+    
+    for node in NON_BASE_SYNTAX_NODES {
+      FunctionDecl("""
+        /// Implementation detail of visit(_:). Do not call directly.
+        private func visitImpl\(raw: node.name)(_ data: SyntaxData) -> Syntax {
+          let node = \(raw: node.name)(data)
+          // Accessing _syntaxNode directly is faster than calling Syntax(node)
+          visitPre(node._syntaxNode)
+          defer { visitPost(node._syntaxNode) }
+          if let newNode = visitAny(node._syntaxNode) { return newNode }
+          return Syntax(visit(node))
+        }
+        """)
+      }
+    
+    FunctionDecl("""
+      /// Implementation detail of visit(_:). Do not call directly.
+      private func visitImplTokenSyntax(_ data: SyntaxData) -> Syntax {
+        let node = TokenSyntax(data)
+        // Accessing _syntaxNode directly is faster than calling Syntax(node)
+        visitPre(node._syntaxNode)
+        defer { visitPost(node._syntaxNode) }
+        if let newNode = visitAny(node._syntaxNode) { return newNode }
+        return Syntax(visit(node))
+      }
+      """)
+        
+    IfConfigDecl(
+      leadingTrivia: .blockComment("""
+        // SwiftSyntax requires a lot of stack space in debug builds for syntax tree
+        // rewriting. In scenarios with reduced stack space (in particular dispatch
+        // queues), this easily results in a stack overflow. To work around this issue,
+        // use a less performant but also less stack-hungry version of SwiftSyntax's
+        // SyntaxRewriter in debug builds.
+        """) + .newline,
```

Comment:
Again, I think these lines need to be individual `.blockComment` with `.newline` for correct indentation.